packages/xo-server/src/xo-mixins/authentication.mjs

```diff
@@ -50,13 +50,28 @@ export default class {
     })
 
     // Token authentication provider.
-    this.registerAuthenticationProvider(async ({ token: tokenId }) => {
+    this.registerAuthenticationProvider(async ({ token: tokenId }, { ip } = {}) => {
       if (!tokenId) {
         return
       }
 
       try {
         const token = await app.getAuthenticationToken(tokenId)
+
+        let { last_uses = {} } = token
+        last_uses[ip] = { timestamp: Date.now() }
+
+        const MAX_LAST_USE_ENTRIES = 10
+        if (Object.keys(last_uses).length > MAX_LAST_USE_ENTRIES) {
+          const entries = Object.entries(last_uses).sort((a, b) => b[0].timestamp - a[0].timestamp)
```

Comment:
Index `0`, is supposed to be the key, no?
```js
Object.entries(last_uses) // [[<ip>, {...}], [<ip>, {...}]]
```

Reply:
:facepalm: